components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/endpoint/registry/util/EndpointRegistryUtil.java

```diff
@@ -32,4 +46,24 @@ public static void raiseResourceAlreadyExistsException(String msg) throws
         throw new EndpointRegistryResourceAlreadyExistsException(msg);
     }
 
+    public static boolean isValidDefinition(URL definitionURL, String definitionType)
+            throws DefinitionValidationException {
+        DefinitionValidator definitionValidator = definitionValidatorMap.get(definitionType);
+        if (definitionValidator != null) {
+            return definitionValidator.validate(definitionURL);
+        }
+        throw new DefinitionValidationException("No Definition Validator found for the given definition type: '"
```

Comment:
Let the exception come through the validation.

Reply:
**throw new DefinitionValidationException(..)** is thrown if definitionValidator is null. So, cannot throw from validate method